lib/hexpm_web/controllers/package_report_controller.ex

```diff
@@ -71,20 +72,34 @@ defmodule HexpmWeb.PackageReportController do
 
   def show(conn, params) do
     report = PackageReports.get(params["id"])
-    to_moderator = Users.has_role(conn.assigns.current_user, "moderator")
+    for_moderator = Users.has_role(conn.assigns.current_user, "moderator")
+    user = conn.assigns.current_user
 
-    if report == nil or
-         (report.state == "to_accept" and !to_moderator) do
+    if report == nil do
       conn
       |> put_flash(:error, @report_not_accessible)
       |> put_status(400)
       |> redirect(to: Routes.package_path(HexpmWeb.Endpoint, :index))
     else
+      {show_accept, show_reject, show_solve} =
+        case report.state do
+          "to_accept" -> {true, true, false}
+          "accepted" -> {false, true, true}
+          _ -> {false, false, false}
+        end
```

Comment:
What do you think about moving this logic to the template?

Instead of `<%= if @show_reject do %>`, do `<%= if @report.state in ["to_accept", "accepted"] do %>`.

Reply:
I think `no_posible_transition` and `no_possible_transition_msg` can also be moved to the template.